### PR TITLE
feat: implement PEBBLE_IMPORT to load layers from other locations

### PR DESCRIPTION
**Pull request by hpidcock**
This PR proposes a way for Pebble to load layers from multiple sources first, before then loading from the `$PEBBLE` directory. This is required when pebble is run as a non-root user in a container with pebble layers already defined at the default `$PEBBLE` location of `/var/lib/pebble/default`.

By setting `$PEBBLE` to a directory writeable by a non-root user, while also setting `$PEBBLE_IMPORT` to `/var/lib/pebble/default`, the pebble daemon is able to load layers previously defined in the default directory by a container image author.

`$PEBBLE_IMPORT` can be a colon (`:`) delimited ordered list of paths to directories in the format of the pebble data directory (`$PEBBLE`). Each directory will have a `layers/` sub-directory.

It is expected that the following pebble invocation, 
`PEBBLE=/var/lib/pebble/charm PEBBLE_IMPORT=/var/lib/pebble/default:/var/lib/pebble/intermediate
pebble run;` would load layers like so:
- /var/lib/pebble/default/layers/001-my-first-layer.yaml
- /var/lib/pebble/default/layers/002-my-second-layer.yaml
- /var/lib/pebble/intermediate/layers/001-extra-first-layer.yaml
- /var/lib/pebble/intermediate/layers/002-extra-second-layer.yaml
- /var/lib/pebble/charm/layers/001-jujus-first-layer.yaml
- /var/lib/pebble/charm/layers/002-jujus-second-layer.yaml

NOTE: order numbers can be re-used globally, but labels must be unique globally.

For tooling to determine where configuration originated from, see https://github.com/canonical/pebble/pull/342

[JU090](https://docs.google.com/document/d/1NWV4QsYq1NldS_V_YlafpJQyNAV-WS46VsIKGBrin_s/edit)

### CODE REVIEW

File: internals/cli/cli.go
Comment (benhoyt) [inline internals/cli/cli.go:366]: I realise this is an internal function, but maybe it's time to have this return a little `type envPaths struct { ... }` with named fields, rather than relying on the order of return arguments.

File: internals/cli/cmd_enter_test.go
Comment (benhoyt) [inline internals/cli/cmd_enter_test.go:155]: What's this change for?

File: internals/overlord/overlord.go
Comment (benhoyt) [inline internals/overlord/overlord.go:63]: Naming bikeshed: I know we have the "Pebble" prefix on the `PebbleDir` field, but I think just `ImportDirs` would be fine here (like on daemon.Options).

File: internals/overlord/servstate/manager.go
Comment (benhoyt) [inline internals/overlord/servstate/manager.go:63]: This is now a 7-param function ... time to use an Options struct? (It's in `internals` so we can change the API.)

File: internals/plan/plan.go
Comment (benhoyt) [inline internals/plan/plan.go:56]: Is this used?

File: internals/plan/plan.go
Comment (benhoyt) [inline internals/plan/plan.go:1025]: Seems like this should not be exported (`layerReader`).

File: internals/plan/plan.go
Comment (benhoyt) [inline internals/plan/plan.go:1046]: Why can't we operate on `lr.labels` directly below, instead of creating a copy?

File: internals/plan/plan.go
Comment (benhoyt) [inline internals/plan/plan.go:1066]: Could use the newer `os.ReadFile` while we're in here (now that we're on Go 1.20; ioutil was deprecated in 1.16).

File: internals/plan/plan.go
Comment (benhoyt) [inline internals/plan/plan.go:1099]: Could just update this in the loop as we go? Would avoid the `if len(layers) > 0`.

File: internals/plan/plan.go
Comment (benhoyt) [inline internals/plan/plan.go:1104]: Maybe `ReadAll` to indicate it reads from all the places.

File: internals/plan/plan.go
Comment (benhoyt) [inline internals/plan/plan.go:1108]: Hmm, this reads a bit oddly. It happens first, but it's at the end of the doc comment.

Could we just change the parameters to take a single `dirs []string` and have the caller append dir to the list it passes as the (single) argument?

**Review by benhoyt — COMMENTED** [review]
Looking good. Just some minor comments -- nothing structural.

File: internals/cli/cli.go
Comment (hpidcock) [reply internals/cli/cli.go:366]: sounds good.

File: internals/overlord/servstate/manager.go
Comment (hpidcock) [reply internals/overlord/servstate/manager.go:63]: Great idea, I've gone ahead and simplified a lot of this.

File: internals/overlord/overlord.go
Comment (hpidcock) [reply internals/overlord/overlord.go:63]: I've done this and also brought PebbleDir in line with other areas just referring to it as Dir.

File: internals/plan/plan.go
Comment (hpidcock) [reply internals/plan/plan.go:56]: No, not yet, will remove in this PR.

File: internals/plan/plan.go
Comment (hpidcock) [reply internals/plan/plan.go:1025]: sounds good.

File: internals/plan/plan.go
Comment (hpidcock) [reply internals/plan/plan.go:1046]: As discussed, this was to simplify duplicate label errors. Since we probably want the full path anyway, I've simplified this.

File: internals/plan/plan.go
Comment (hpidcock) [reply internals/plan/plan.go]: Fantastic idea, will make the change.

**Review by benhoyt — APPROVED** [review]
Looks good to me now, thanks! Let's just wait for Gustavo's final review on [JU090](https://docs.google.com/document/d/1NWV4QsYq1NldS_V_YlafpJQyNAV-WS46VsIKGBrin_s/edit) before merging.

**Pull request closed without merging**
